File: Peperino/Dtos/Inventory/InventoryOutDto.cs
using Peperino.Dtos.CheckList;
using Peperino.Dtos.UserGroup;
using System.ComponentModel.DataAnnotations;

namespace Peperino.Dtos.Inventory
{
    public class InventoryOutDto
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public RoomOutDto Room_Inventory { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Slug { get; set; } = string.Empty;

        [Required]
        public List<BaseCheckListItemOutDto> Entities { get; set; } = new List<BaseCheckListItemOutDto>();
    }
}

File: peperino.ui/src/lib/api/models/InventoryOutDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { AccessLevel } from './AccessLevel';
import type { BaseCheckListItemOutDto } from './BaseCheckListItemOutDto';
import type { RoomOutDto } from './RoomOutDto';

export type InventoryOutDto = {
    id: number;
    room_Inventory: RoomOutDto;
    name: string;
    slug: string;
    entities: Array<BaseCheckListItemOutDto>;
    accessLevel: AccessLevel;
};
